biography/models.py

```diff
@@ -1,125 +1,114 @@
+    modified = models.DateTimeField(auto_now=True)
+    first_name = models.TextField()
+    last_name = models.TextField()
     date_of_birth = models.DateTimeField(
         _("Date of birth"), blank=True, null=True)
     date_of_passing = models.DateTimeField(
         _("Date of passing"), blank=True, null=True)
     place_of_birth = models.TextField(_("Place of birth"))
     place_of_passing = models.TextField(
         _("Place of passing"), blank=True, null=True)
-    user = models.ForeignKey(
-        User, verbose_name=_("Person"), on_delete=models.CASCADE,
-        help_text=_('all other personal data is stored on the user'))
-    editors = models.ManyToManyField(User, verbose_name=_("Editors"),
-        related_name='editing_bios',
-        help_text=_('the good people that can edit this biography'))
-    # creator is critical when we have issues of ownership and liability
+    editors = models.ManyToManyField(User,
+                                     verbose_name=_("Editors"),
+                                     related_name='editing_bios',
+                                     help_text=_('the good people that can edit this biography'))
+    periods = models.ManyToManyField(Period,
+                                     through='periods')
+    contentAtoms = models.ManyToManyField(ContentAtom,
+                                          related_name='content_atoms')
     creator = models.ForeignKey(
         User, on_delete=models.CASCADE, related_name='created_bios')
-    is_public = models.BooleanField(_("Public"), default=False,
-                    help_text=_('Is it public?'))
     created = models.DateTimeField(auto_now_add=True)
-    modified = models.DateTimeField(auto_now=True)
 
-    class Meta:
+        class Meta:
         verbose_name = _("Biography")
         verbose_name_plural = _("Biographies")
 
     @property
+    def get_username(self):
+        return f'{first_name}{last_name}'
+
     def is_dead(self):
         return not self.date_of_death
 
     def get_absolute_url(self):
+        # Why are two options needed here?
         ''' the url of the biography. user the uuid for the living,
             username for the dead.
         '''
         if self.is_dead:
-            return "/" + self.username
+            return self.get_username()
         else:
             return "/" + uuid
 
     def __str__(self):
-        return 'The biography of {}'.format(self.user.get_full_name())
+        return f'The Biography of {self.first_name}{self.last_name}'
+
+
+class Period(models.Model):
+    ''' a meaningful period in a life '''
+    person = models.ForeignKey(
+        Person, verbose_name=_("Person"), on_delete=models.CASCADE)
+    title = models.CharField(
+        max_length=80, help_text=_('unique title for period'))
+    start_date = models.DateField(_("Start Date"))
+    end_date = models.DateField(_("End Date"))
+    contentAtoms = models.ManyToManyField(
```

Comment:
These are content atom that were manually added. Period content include this and all the ContentAtoms taking during the period (based on date)